plugins/woocommerce/src/Internal/Admin/Settings.php

```diff
@@ -207,10 +212,14 @@ public function add_component_settings( $settings ) {
 		// WooCommerce Branding is an example of this - so pass through the translation of
 		// 'WooCommerce' to wcSettings.
 		$settings['woocommerceTranslation'] = __( 'WooCommerce', 'woocommerce' );
-		// We may have synced orders with a now-unregistered status.
-		// E.g An extension that added statuses is now inactive or removed.
-		if ( PageController::is_admin_page() ) {
-			$settings['unregisteredOrderStatuses'] = $this->get_unregistered_order_statuses();
+
+		if ( PageController::is_admin_page() && Features::is_enabled( 'analytics' ) ) {
+			$is_analytics_page = str_starts_with( wc_clean( wp_unslash( $_GET['path'] ?? '' ) ), '/analytics/' );
```

Comment:
Unfortunately, these settings need to be hydrated on all wc-admin page loads and not just those with `analytics` in the path.

You can see this breaking if we start on `/wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Fsettings` you should see unregistered statuses:
<img width="316" alt="Screenshot 2025-02-28 at 10 41 56 AM" src="https://github.com/user-attachments/assets/c95dac00-5999-41bd-b1f6-e3ad6e78080f" />

However, if we start on WooCommerce -> Home (refresh the page there) and navigate to Analytics -> Settings, we'll see those statuses are missing:

<img width="340" alt="Screenshot 2025-02-28 at 10 45 00 AM" src="https://github.com/user-attachments/assets/adb779ba-d7ce-4e95-8b72-f337faf8c911" />

Reply:
Re-evaluated how to play nicely with hydration and re-iterating with long-living transient (1 week -> 1 month), which gets actualized on order status change.